src/canary.rs

```diff
@@ -339,22 +333,19 @@ mod measure_subroutine {
     /// Read out result from register r0 and process it to get lowest touched byte.
     ///
     /// Happens after the subroutine finishes.
-    fn get_result(core: &mut Core) -> Result<Option<u32>, probe_rs::Error> {
+    fn get_result(core: &mut Core) -> Result<u32, probe_rs::Error> {
         // get the address of the lowest touched 4-byte-word
-        let word_addr = match core.read_core_reg(RegisterId(0))? {
-            0 => return Ok(None),
-            n => n,
-        };
+        let word_addr = core.read_core_reg(RegisterId(0))?;
 
         // take a closer look at word, to get address of lowest touched byte
         let offset = core
-            .read_word_32(word_addr as u64)?
+            .read_word_32(word_addr)?
             .to_le_bytes()
             .into_iter()
             .position(|b| b != CANARY_U8)
             .unwrap();
```

Comment:
(edit: Github is pointing at a not very helpful line of code - the comment applies to the whole function)

Reply:
I guess it is possible because we are dealing with hardware here xD

But in that case I'd expect this function to return `0` which then will get reported as `(HOST) INFO  program has used at least 0.0/254.93 KiB (0.0%) of stack space`. Which I think is fair.

Reply:
Returning 0 is reasonable. Unwrapping is probably not reasonable.

Reply:
I did change it back to returning an `Option`. And I replaced the `.unwrap()` with an `.expect()`.

Reply:
I think I might suggest the other way? Return a u32, but replace the unwrap with an `unwrap_or_else(...)` so you do not panic if the every byte in the stack is equal to the CANARY_U8 but instead cause the function to return "0 bytes of stack used".

Reply:
The u32 is actually the address and not the amount of memory used. The address is only calculated in `fn measure`. So the function kinda reports if we found a touched address, or if we found none. So the Option kinda makes sense.

Reply:
Oh, I see. This particular line of code is looking for the first non-CANARY_U8 byte within a word that has already been checked as being != CANARY_U32. So the unwrap on line 341 is probably appropriate (but could be documented to avoid confusion). Sorry for the brain fog.

Reply:
No worries, haha. I did confuse it as well.

I tried to make it more clear by adding the error message. Do you think this makes it better?

https://github.com/knurling-rs/probe-run/blob/30f2284d080a4a1d2d57894a3172e6397c6d63d1/src/canary.rs#L380